Rubberduck.Parsing/VBA/RubberduckParserState.cs

```diff
@@ -335,15 +390,15 @@ private void RefreshProject(string projectId)
         public event EventHandler<ParserStateEventArgs> StateChanged;
 
         private int _stateChangedInvocations;
-        private void OnStateChanged(object requestor, CancellationToken token, ParserState state = ParserState.Pending)
+        private void OnStateChanged(object requestor, CancellationToken token, ParserState state = ParserState.Pending, ParserState oldStatus = ParserState.Pending)
```

Comment:
I do not really like that the `oldState` is optional. Would it be possible to make that not optional and do the same for the `state`. I think we should always be explicit about what the state change is.